public/assets/index.js

```diff
@@ -120,28 +120,49 @@ function handleDeltaUpdate(message) {
     if (message["name"] in block || block.includes(message["name"])) {
         return;
     }
-
-    var text = `<strong>${message["name"]}</strong> has gone ${mapState(message["state"])}! (${message["section"]})`;
-
+    
+    const subredditElement = document.getElementById(message["name"]);
+    const subredditStatusElement = subredditElement.querySelector("p");
+
+    const mappedState = mapState(message["state"]);
+    const prevState = message["previous_state"];
+    
+    // create status update string
+    var text = `<strong>${message["name"]}</strong> (${message["section"]})<br>${prevState} → <strong>${mappedState}</strong>`;
+    
+    switch (message["state"]) {
+        case "PRIVATE":
+            if (prevState != "RESTRICTED") text += "!";
+            break;
+        case "RESTRICTED":
+            if (prevState != "PRIVATE") text += "!";
+            break;
+        case "PUBLIC":
+            text += " :(";
+            break;
+        default:
+            break;
+    }
+    
     // Send out status update for people not in large counter mode.
     newStatusUpdate(text, function () {
-        doScroll(document.getElementById(message["name"]));
+        doScroll(subredditElement);
     }, ["n" + sectionBaseName(message["section"])]);
 
     // Update state in current view if present.
-    if (document.getElementById(message["name"]) != null) {
-        document.getElementById(message["name"]).querySelector("p").innerHTML = mapState(message["state"]);
+    if (subredditElement != null) {
+        subredditStatusElement.innerHTML = mappedState;
         for (i of ["private", "public", "restricted", "unknown"]) {
-            document.getElementById(message["name"]).classList.remove(`subreddit-${i}`)
+            subredditElement.classList.remove(`subreddit-${i}`)
         }
-        document.getElementById(message["name"]).classList.add(`subreddit-${mapState(message["state"])}`)
+        subredditElement.classList.add(`subreddit-${mappedState}`)
     }
 
     switch (message["state"]) {
         case "PRIVATE":
         case "RESTRICTED":
             audioSystem.play("privated");
-            switch (subreddit["previous_state"]) {
+            switch (message["previous_state"]) {
```

Comment:
i edited `subreddit` to `message` here as i couldn’t find a `subreddit` variable defined in the function. hopefully that was correct (lmk if not)